gcloud/translate/client.py

```diff
@@ -0,0 +1,199 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Client for interacting with the Google Cloud Translate API."""
+
+
+import httplib2
+import six
+
+from gcloud._helpers import _to_bytes
+from gcloud.translate.connection import Connection
+
+
+ENGLISH_ISO_639 = 'en'
+"""ISO 639-1 language code for English."""
+
+
+def _zip_assert(values1, values2):
+    """Zips two iterables and asserts they are the same length.
+
+    :type values1: list
+    :param values1: Iterable (with a length) to be zipped.
+
+    :type values2: list
+    :param values2: Iterable (with a length) to be zipped.
+
+    :rtype: iterator
+    :returns: Zip iterator for each pair of values.
+    :raises: :class:`ValueError <exceptions.ValueError>` if the number of
+             values in each list is not the same.
+    """
+    if len(values1) != len(values2):
+        raise ValueError('Expected iterations to have same length',
+                         values1, values2)
+
+    return six.moves.zip(values1, values2)
+
+
+class Client(object):
+    """Client to bundle configuration needed for API requests.
+
+    :type key: str
+    :param key: The key used to send with requests as a query
+                parameter.
+
+    :type http: :class:`httplib2.Http` or class that defines ``request()``.
+    :param http: (Optional) HTTP object to make requests. If not
+                 passed, an :class:`httplib.Http` object is created.
+    """
+
+    def __init__(self, key, http=None):
+        self.key = key
+        if http is None:
+            http = httplib2.Http()
+        self.connection = Connection(http=http)
+
+    def get_languages(self, target_language=None):
+        """Get list of supported languages for translation.
+
+        Response
+
+        See: https://cloud.google.com/translate/v2/\
+        discovering-supported-languages-with-rest
+
+        :type target_language: str
+        :param target_language: (Optional) The language used to localize
+                                returned language names.
+
+        :rtype: list
+        :returns: List of dictionaries. Each dictionary contains a supported
+                  ISO 639-1 language code (using the dictionary key
+                  ``language``). If ``target_language`` is passed, each
+                  dictionary will also contain the name of each supported
+                  language (localized to the target language).
+        """
+        query_params = {'key': self.key}
+        if target_language is not None:
+            query_params['target'] = target_language
+        response = self.connection.api_request(
+            method='GET', path='/languages', query_params=query_params)
+        return response.get('data', {}).get('languages', ())
+
+    def detect_language(self, *values):
```

Comment:
I'm -1 on special-casing the single-value bits.  After all, the translate docs all have the following at the top of the page:

> Google Translate API is a paid enterprise service for translating large amounts of text. For website translations, we encourage you to use the Google Website Translator gadget.